### PR TITLE
Include Consumed Units in TUV

### PR DESCRIPTION
**Pull request by ron-murhammer**
Addresses: https://forums.triplea-game.org/topic/159/tuv-for-units-that-have-consumesunits-property

Adds value of all consumeUnits for each unit. For example on Global 1940, Factory_Upgrade is worth 32 (12+20). It does this recursively and checks to ensure to count each unit only once if there are cycles (unitA -> unitB -> unitA = unitA + unitB).

### CODE REVIEW

File: src/main/java/games/strategy/triplea/ui/TripleAFrame.java [hunk 1]
```diff
@@ -1048,7 +1048,7 @@ public void run() {
         for (final Entry<Territory, Collection<Unit>> entry : possibleUnitsToAttack.entrySet()) {
           final List<Unit> units = new ArrayList<>(entry.getValue());
           Collections.sort(units,
-              new UnitBattleComparator(false, TuvUtils.getCostsForTuvForAllPlayersMergedAndAveraged(data),
+              new UnitBattleComparator(false, TuvUtils.getCostsForTuv(units.get(0).getOwner(), data),
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/TripleAFrame.java]: Is there a possibility that `units` can be empty here?

File: src/main/java/games/strategy/triplea/util/TuvUtils.java [hunk 2]
```diff
@@ -129,6 +136,20 @@ private TuvUtils() {}
     return costs;
   }
 
+  private static int getTotalTuv(UnitType unitType, IntegerMap<UnitType> costs, Set<UnitType> alreadyAdded) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/util/TuvUtils.java]: Nit: Running Clean Up here will mark the method parameters `final`.

**Comment by RoiEXLab** [comment]
I restarted the push build... Fixed the issue temporarily

File: src/main/java/games/strategy/triplea/ui/TripleAFrame.java (hunk 1 above)
Comment (ron-murhammer) [reply src/main/java/games/strategy/triplea/ui/TripleAFrame.java]: Nope. There is only 1 call path to this and it only inserts into the map if the collection of units isn't empty.

File: src/main/java/games/strategy/triplea/util/TuvUtils.java (hunk 2 above)
Comment (ron-murhammer) [reply src/main/java/games/strategy/triplea/util/TuvUtils.java]: Good catch. Updated.

**Comment by RoiEXLab** [comment]
Push build failing again, not to bother about, merging!

**Pull request merged**